crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -3363,8 +3414,6 @@ impl<'db> TypeInferenceBuilder<'db> {
                             bindings: _,
                             errors,
                         } => {
-                            // TODO: Remove the `Vec::from` call once we use the Rust 2024 edition
-                            //  which adds `Box<[T]>::into_iter`
```

Comment:
...did they remove it again? :P

Reply:
Why am I so terrible lately with rebasing/stashing my changes. No, I figured out how to avoid the `Vec::from` but the change got lost. Thanks for catching this.